conf/tracing.go

```diff
@@ -1,38 +1,180 @@
+	"context"
 	"fmt"
+	"github.com/sirupsen/logrus"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/exporters/prometheus"
+	"go.opentelemetry.io/otel/metric/global"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/metric/controller/basic"
+	controller "go.opentelemetry.io/otel/sdk/metric/controller/basic"
+	"go.opentelemetry.io/otel/sdk/metric/selector/simple"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.10.0"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"net/http"
+	"time"
 
-	"github.com/opentracing/opentracing-go"
-	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/opentracer"
-	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	processor "go.opentelemetry.io/otel/sdk/metric/processor/basic"
+)
+
+type Exporter = string
+
+const (
+	OtlpGrpcExporter   Exporter = "otlpgrpc"
+	OtlpHttpExporter   Exporter = "otlphttp"
+	PrometheusExporter Exporter = "prometheus"
+	NoopExporter       Exporter = "noop"
+	DefaultExporter    Exporter = "default"
 )
 
 type TracingConfig struct {
-	Enabled     bool `default:"false"`
-	Host        string
-	Port        string
-	ServiceName string `default:"gotrue" split_words:"true"`
+	Host        string   `default:"127.0.0.1"`
+	Port        string   `default:"2222"`
+	Exporter    Exporter `default:"noop"`
+	ServiceName string   `default:"gotrue" split_words:"true"`
 	Tags        map[string]string
+
+	// INTERNAL, used to start in `api.go`
+	OtlpMetricExporter *otlpmetric.Exporter            `ignore:"true"`
+	TracingShutdown    func(ctx context.Context) error `ignore:"true"`
+	ContextCancel      func()                          `ignore:"true"`
 }
 
 func (tc *TracingConfig) tracingAddr() string {
 	return fmt.Sprintf("%s:%s", tc.Host, tc.Port)
 }
 
 func ConfigureTracing(tc *TracingConfig) {
-	var t opentracing.Tracer = opentracing.NoopTracer{}
-	if tc.Enabled {
-		tracerOps := []tracer.StartOption{
-			tracer.WithServiceName(tc.ServiceName),
-			tracer.WithAgentAddr(tc.tracingAddr()),
+	var t trace.TracerProvider
+	var c *controller.Controller
+	var customAttributes []attribute.KeyValue
+	for k, v := range tc.Tags {
+		customAttributes = append(customAttributes, attribute.Key(k).String(v))
+	}
+	customAttributes = append(customAttributes, semconv.ServiceNameKey.String(tc.ServiceName))
+	resourceAttributes := resource.WithAttributes(customAttributes...)
+
+	switch tc.Exporter {
+	case OtlpGrpcExporter:
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
```

Comment:
Http is a push model, while GRPC is a 2 way sustained connection so if there isn't a timeout it could block startup when it should just be throwing the error earlier